numpy/lib/tests/test_nanfunctions.py

```diff
@@ -388,6 +410,101 @@ def test_empty(self):
             assert_equal(res, tgt)
 
 
+class TestNanFunctions_CumSumProd(TestCase, SharedNanFunctionsTestsMixin):
+
+    nanfuncs = [np.nancumsum, np.nancumprod]
+    stdfuncs = [np.cumsum, np.cumprod]
+
+    def test_allnans(self):
+        # Check for FutureWarning
+        for f, tgt_value in zip(self.nanfuncs, [0, 1]):
+            with warnings.catch_warnings(record=True) as w:
```

Comment:
what is assert no warnings anyway? I personally want warnings to be errors always inside the test (unless it is release version) to be honest. Asserting no warning seems like a broken concept. If there are warnings, you have to catch them, if there are not, you don't. If you have to mix it, you are screwed, unless you merge my new PR, heh....